src/MoBi.Core/Domain/Repository/BuildingBlockRepository.cs

```diff
@@ -1,43 +1,83 @@
+      ExpressionProfileBuildingBlock ExpressionProfileByName(string buildingBlockName);
    }
 
    public class BuildingBlockRepository : IBuildingBlockRepository
    {
       private readonly IMoBiProjectRetriever _projectRetriever;
-      
+
       public BuildingBlockRepository(IMoBiProjectRetriever projectRetriever)
       {
          _projectRetriever = projectRetriever;
       }
 
       public IEnumerable<IBuildingBlock> All()
       {
-         return All<IBuildingBlock>();
+         return IndividualsCollection
+            .Concat<IBuildingBlock>(ExpressionProfileCollection)
+            .Concat(moduleBuildingBlocks()).ToList();
       }
 
-      public IEnumerable<TBuildingBlock> All<TBuildingBlock>() where TBuildingBlock : IBuildingBlock
+      private IEnumerable<IBuildingBlock> moduleBuildingBlocks()
       {
-         var currentProject = _projectRetriever.Current;
-         if (currentProject == null)
-            return new List<TBuildingBlock>();
+         return _projectRetriever.Current.Modules.SelectMany(x => x.BuildingBlocks);
+      }
+
+      private IReadOnlyList<T> get<T>()
+      {
+         return All().OfType<T>().ToList();
+      }
+
+      public IReadOnlyList<MoleculeBuildingBlock> MoleculeBlockCollection => get<MoleculeBuildingBlock>();
+
+      public IReadOnlyList<MoBiReactionBuildingBlock> ReactionBlockCollection => get<MoBiReactionBuildingBlock>();
 
-         return currentProject.AllBuildingBlocks().OfType<TBuildingBlock>().Concat(moduleBuildingBlocks<TBuildingBlock>(currentProject));
+      public IReadOnlyList<ExpressionProfileBuildingBlock> ExpressionProfileCollection => _projectRetriever.Current.ExpressionProfileCollection;
+
+      public IReadOnlyList<IndividualBuildingBlock> IndividualsCollection => _projectRetriever.Current.IndividualsCollection;
+
+      public IReadOnlyList<PassiveTransportBuildingBlock> PassiveTransportCollection => get<PassiveTransportBuildingBlock>();
+
+      public IReadOnlyList<MoBiSpatialStructure> SpatialStructureCollection => get<MoBiSpatialStructure>();
+
+      public IReadOnlyList<ObserverBuildingBlock> ObserverBlockCollection => get<ObserverBuildingBlock>();
+
+      public IReadOnlyList<EventGroupBuildingBlock> EventBlockCollection => get<EventGroupBuildingBlock>();
+
+      public IReadOnlyList<InitialConditionsBuildingBlock> InitialConditionBlockCollection => get<InitialConditionsBuildingBlock>();
+
+      public IReadOnlyList<ParameterValuesBuildingBlock> ParametersValueBlockCollection => get<ParameterValuesBuildingBlock>();
+
+      public IndividualBuildingBlock IndividualByName(string buildingBlockName)
+      {
+         return IndividualsCollection.FindByName(buildingBlockName);
       }
 
-      private static IEnumerable<TBuildingBlock> moduleBuildingBlocks<TBuildingBlock>(MoBiProject currentProject) where TBuildingBlock : IBuildingBlock
+      public ExpressionProfileBuildingBlock ExpressionProfileByName(string buildingBlockName)
```

Comment:
We are using the ByName when configuring a simulation. In that case we have some clones of these building blocks and we want to go back to the like-named building block from the project if the user reconfigures... at least for now.

We do the same for Modules, but that code lives in the MoBiProject because there's no 'ModuleRepository' in the same way as a BuildingBlockRepository.

Due to you pointing this code out, I found that these are not being used, and the same code was still in the project. I am removing it from the project and using this - it was my original intention.